python/ray/serve/_private/application_state.py

```diff
@@ -160,9 +160,9 @@ def update(self):
                     return
                 try:
                     ray.get(finished[0])
-                except RayTaskError:
+                except RayTaskError as e:
                     self.status = ApplicationStatus.DEPLOY_FAILED
-                    self.app_msg = f"Deployment failed:\n{traceback.format_exc()}"
+                    self.app_msg = f"Deployment failed:\n{str(e)}"
```

Comment:
how does this change the formatting & why is it required?

Reply:
I added an example of how it changes the formatting in the PR description. I haven't dug into why yet (only saw this [custom formatting function](https://github.com/ray-project/ray/blob/a892241ca7574af47f278a667e6493a4b03686d7/python/ray/exceptions.py#L163)), I can look into it more later today.

Reply:
oh -- thanks for doing that :)

that's fine, let's leave a comment indicating that it is important and shouldn't be changed to `traceback.format_exc()` (which is more standard in my experience)

Reply:
Sg, will do!